### ISSUE TITLE
Ensure all [entity]_save() functions return SAVED_NEW or SAVED_UPDATED with proper docs

### ISSUE THREAD

**Issue by sirkitree** [backdrop/backdrop-issues#983]
The API states
> Return value
> The saved $user object upon successful save or FALSE if the save failed.

However, when creating a new user programatically, I'm finding that this is not true. I don't get anything back at all it seems.

Here is the code I've tried:

```php
/**
 * @param $migration_data: data from migration to create a new user.
 * @return $user object
 */
function my_create_user_function($migration_data) {
  $account = new User();
  $account->is_new = TRUE;
  $account->name = $data->username;
  $account->mail = $data->email;
  $account->pass = user_password(8);
  $account->status = $data->enabled;
  $account->init = $data->email;
  $account->signature_format = 'plain_text';
  $account->created = $data->signup_date;
  $account->roles = array(BACKDROP_AUTHENTICATED_ROLE => 'authenticated');
  return user_save($account);
}
```

Am I doing something wrong, or should `user_save()` actually be returning the new user object?

**Comment by quicksketch** [comment]
> Am I doing something wrong, or should user_save() actually be returning the new user object?

I believe what `user_save()` returns is now the same as `node_save()` or [any other entity save](https://api.backdropcms.org/api/backdrop/core!modules!entity!entity.class.inc/function/EntityInterface%3A%3Asave/1), that is it returns either SAVED_NEW or SAVED_UPDATED constants based on the operation.

What should happen is that the $account variable is modified by reference, so you should be able to get `account->uid` right after saving the user.

You can also (if desired), use `$account->save()` instead of `user_save($account)`, but it's all the same.

So bottom line, I think we have a documentation bug on the return value of `user_save()`.

**Comment by sirkitree** [comment]
Got it, thank you!

**Comment by rszrama** [comment]
Funny. Yes, we do need to update the user_save() document so it indicates the proper return value. However, node_save() actually doesn't return anything right now, even though $node->save() does. :stuck_out_tongue:

**Comment by rszrama** [comment]
Ok, on just a small sample, I see the inconsistency in other modules as well (e.g. comment_save() has no return value but the taxonomy_*_save() functions do). What do you think of repurposing this issue to standardize all these functions invoking $entity->save(); to return the return value of that operation? And updating the documentation for all of these functions in the process (e.g. return parameter, proper class for the entity parameter, etc.)?

**Comment by quicksketch** [comment]
> However, node_save() actually doesn't return anything right now, even though $node->save() does. 

Doh!

> What do you think of repurposing this issue to standardize all these functions invoking $entity->save(); to return the return value of that operation?

Yes please.

One thing I'd like to get in 2.x is having all entity *types* become true objects as well. Because right now $term is a real object but $vocabulary is not. But that's not for this issue. I'd love to make sure that term/user/node/file functions at least all behave consistently, as the underlying entity objects already do.

**Comment by quicksketch** [comment]
Thanks @rszrama for fixing this issue! Merged in https://github.com/backdrop/backdrop/pull/953 to 1.x and 1.1.x.
